ExampleMod/Content/TileEntities/BasicTileEntity.cs

```diff
@@ -0,0 +1,239 @@
+﻿using Microsoft.Xna.Framework;
+using System;
+using System.IO;
+using Terraria;
+using Terraria.Audio;
+using Terraria.DataStructures;
+using Terraria.ID;
+using Terraria.Localization;
+using Terraria.ModLoader;
+using Terraria.ModLoader.IO;
+using Terraria.ObjectData;
+
+namespace ExampleMod.Content.TileEntities
+{
+	// This file contains all the code necessary for a basic Tile Entity. It is intended to showcase a fully working example while being simple enough to be easily followed and adapted if needed. Follow the https://github.com/tModLoader/tModLoader/wiki/Basic-Tile-Entity guide for more information, this example is essentially the result of following the guide.
+
+	// A tile entity is an object bound to a placed tile to allow it additional capabilities normal tiles are unable to have. For example, normal tiles can't store data and can't run code every game update. This is the main purpose of tile entities.
+	// This file contains the ModTile, ModItem, and ModTileEntity classes to demonstrate this fully self-contained example:
+	// - ModTileEntity - the tile entity attached to the tile. It provides the additional logic and data storage.
+	// - ModTile - the tile that the ModTileEntity will be attached to.
+	// - ModItem - places the tile.
+
+	// BasicTileEntity is essentially a water barrel that collects water while it is raining. 
+	public class BasicTileEntity : ModTileEntity
+	{
+		private bool syncNeeded; // Used to track when a network sync is needed
+
+		// This water barrel will fill up in 1 minute of rain, or 3600 game updates.
+		private const int MaxFill = 3600;
+		private const int SyncInterval = 3600 / 10;
```

Comment:
Reference `MaxFill` here. Or consider doing `SyncIntervals = 10` and `waterFillLevel * MaxFill / SyncIntervals`